soepy/test/test_unit.py

```diff
@@ -297,7 +325,58 @@ def test_no_children_prob_0():
         covariates,
         child_age_update_rule,
         prob_child,
+        prob_educ_years,
         is_expected=False,
     )
 
     np.testing.assert_equal(sum(df.dropna()["Age_Youngest_Child"] != -1), expected)
+
+
+def test_educ_level_shares():
+    """This test ensures that the shares of individuals with low, middle and high
+    education level in the simulated data frame correspond to the probabilities
+    specified in the init file.
+    """
+
+    constr = dict()
+    constr["AGENTS"] = 10000
+    constr["PERIODS"] = 6
+
+    random_init(constr)
+
+    model_params_df, model_params = read_model_params_init("test.soepy.pkl")
+    model_spec = read_model_spec_init("test.soepy.yml", model_params_df)
+
+    # Set probability of having children to zero for all periods
+    prob_child = np.full(model_spec.num_periods, 0.00)
+
+    # Generate frequencies of different education levels
+    prob_educ_years = np.random.random(3)
+    prob_educ_years /= prob_educ_years.sum()
+    prob_educ_years_list = list(prob_educ_years)
+
+    # Solve
+    states, indexer, covariates, emaxs, child_age_update_rule = pyth_solve(
+        model_params, model_spec, prob_child, is_expected=False,
+    )
+
+    # Simulate
+    df = pyth_simulate(
+        model_params,
+        model_spec,
+        states,
+        indexer,
+        emaxs,
+        covariates,
+        child_age_update_rule,
+        prob_child,
+        prob_educ_years_list,
+        is_expected=False,
+    )
+
+    simulated = (
```

Comment:
Please set up a proper multiindex `identifier` and `period` for the simulated dataframe. 

```
df.set_index(["Identifier", "Period"], inplace=True)
df.sort_index(inplace=True)
```

Then this command becomes

 `df.groupby("Identifier")["Years_of_Education"].first().value_counts(normalize=True)`

which is much more readable.